CMake build fixes by @madebr

diff --git a/CMakeLists.txt b/CMakeLists.txt
index efe857d..5cc32d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -680,7 +680,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libtiff-4.pc
 
 # Includes used by libtiff (and tests)
 if(ZLIB_INCLUDE_DIRS)
-  list(APPEND TIFF_INCLUDES ${ZLIB_INCLUDE_DIRS})
+  list(APPEND TIFF_INCLUDES $<BUILD_INTERFACE:${ZLIB_INCLUDE_DIRS}>)
 endif()
 if(DEFLATE_INCLUDE_DIR)
   list(APPEND TIFF_INCLUDES ${DEFLATE_INCLUDE_DIR})
diff --git a/libtiff/CMakeLists.txt b/libtiff/CMakeLists.txt
index 080685d..0c1607a 100644
--- a/libtiff/CMakeLists.txt
+++ b/libtiff/CMakeLists.txt
@@ -116,9 +116,9 @@ add_library(tiff ${tiff_SOURCES} ${tiff_HEADERS} ${nodist_tiff_HEADERS}
             ${tiff_port_SOURCES} libtiff.def)
 target_include_directories(tiff
     PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-        ${TIFF_INCLUDES}
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>"
+        "$<BUILD_INTERFACE:${TIFF_INCLUDES}>"
 )
 target_link_libraries(tiff ${TIFF_LIBRARY_DEPS})
 set_target_properties(tiff PROPERTIES SOVERSION ${SO_COMPATVERSION})
-- 
1.7.1

