From 36c522699bd8d8b330f61fa2eb10af29f5fdf6b4 Mon Sep 17 00:00:00 2001
From: Yuriy Chernyshov <georgthegreat@gmail.com>
Date: Tue, 5 Sep 2023 09:44:52 +0000
Subject: [PATCH] Fix using __attribute__ libtiff with clang-for-windows

---
 libtiff/tiffio.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libtiff/tiffio.h b/libtiff/tiffio.h
index d6bf0cc..2046054 100644
--- a/libtiff/tiffio.h
+++ b/libtiff/tiffio.h
@@ -261,7 +261,7 @@ typedef struct
 #define LOGLUV_PUBLIC 1
 #endif
 
-#if defined(__GNUC__) || defined(__attribute__)
+#if defined(__GNUC__) || defined(__clang__) || defined(__attribute__)
 #  define TIFF_ATTRIBUTE(x)    __attribute__(x)
 #else
 #  define TIFF_ATTRIBUTE(x) /*nothing*/
