diff -up a/libtiff/tif_dir.c b/libtiff/tif_dir.c
--- a/libtiff/tif_dir.c
+++ b/libtiff/tif_dir.c
@@ -581,7 +581,7 @@ _TIFFVSetField(TIFF* tif, uint32 tag, va_list ap)
 				uint16 v[2];
 				v[0] = (uint16)va_arg(ap, int);
 				v[1] = (uint16)va_arg(ap, int);
-				_TIFFmemcpy(tv->value, &v, 4);
+				_TIFFmemcpy(tv->value, v, 4);
 			}
 
 			else if (fip->field_passcount
diff -up a/libtiff/tif_dirread.c b/libtiff/tif_dirread.c
--- a/libtiff/tif_dirread.c
+++ b/libtiff/tif_dirread.c
@@ -5598,7 +5598,7 @@ ChopUpSingleUncompressedStrip(TIFF* tif)
 	if (rowsperstrip >= td->td_rowsperstrip)
 		return;
 	nstrips64 = TIFFhowmany_64(bytecount, stripbytes);
-	if ((nstrips64==0)||(nstrips64>0xFFFFFFFF)) /* something is wonky, do nothing. */
+	if ((nstrips64==0)||(nstrips64>0xFFFFFFFFULL)) /* something is wonky, do nothing. */
 	    return;
 	nstrips32 = (uint32)nstrips64;
 
diff -up a/libtiff/tif_dirwrite.c b/libtiff/tif_dirwrite.c
--- a/libtiff/tif_dirwrite.c
+++ b/libtiff/tif_dirwrite.c
@@ -1627,7 +1627,7 @@ TIFFWriteDirectoryTagLongLong8Array(TIFF* tif, uint32* ndir, TIFFDirEntry* dir,
 
     for (q=p, ma=value, mb=0; mb<count; ma++, mb++, q++)
     {
-        if (*ma>0xFFFFFFFF)
+        if (*ma>0xFFFFFFFFULL)
         {
             TIFFErrorExt(tif->tif_clientdata,module,
                          "Attempt to write value larger than 0xFFFFFFFF in Classic TIFF file.");
@@ -1685,7 +1685,7 @@ TIFFWriteDirectoryTagIfdIfd8Array(TIFF* tif, uint32* ndir, TIFFDirEntry* dir, ui
 
     for (q=p, ma=value, mb=0; mb<count; ma++, mb++, q++)
     {
-        if (*ma>0xFFFFFFFF)
+        if (*ma>0xFFFFFFFFULL)
         {
             TIFFErrorExt(tif->tif_clientdata,module,
                          "Attempt to write value larger than 0xFFFFFFFF in Classic TIFF file.");
diff -up a/libtiff/tif_strip.c b/libtiff/tif_strip.c
--- a/libtiff/tif_strip.c
+++ b/libtiff/tif_strip.c
@@ -239,7 +239,7 @@ _TIFFDefaultStripSize(TIFF* tif, uint32 s)
 		rows=(uint64)STRIP_SIZE_DEFAULT/scanlinesize;
 		if (rows==0)
 			rows=1;
-		else if (rows>0xFFFFFFFF)
+		else if (rows>0xFFFFFFFFULL)
 			rows=0xFFFFFFFF;
 		s=(uint32)rows;
 	}

